### PR TITLE
Add multi-output regression support for CascadeForestRegressor

### PR DESCRIPTION
**Pull request by Alex-Medium**

### CODE REVIEW

**Comment by xuyxu** [comment]
Thanks for your PR @Alex-Medium, I will take a careful look soon.

**Comment by xuyxu** [comment]
cc @tczhao

This PR extends your contributions on the `CascadeForestRegressor`, will appreciate it very much if you can leave some comments on the code ;-)

**Comment by tczhao** [comment]
LGTM

**Comment by tczhao** [comment]
It'd be good if we can add tests that don't use the 'use_predictor' in tests/test_model_{type}.py

**Comment by Alex-Medium** [comment]
> It'd be good if we can add tests that don't use the 'use_predictor' in tests/test_model_{type}.py

Thanks zhao @tczhao, I will work on this.

File: deepforest/_layer.py [hunk 1]
```diff
@@ -135,10 +135,7 @@ def fit_transform(self, X, y, sample_weight=None):
         n_samples, self.n_features = X.shape
 
         X_aug = []
-        if self.is_classifier:
```
Comment (xuyxu) [inline deepforest/_layer.py:138]: I think there is no need to use the attribute `is_classifier` any more. Could you remove it?

File: deepforest/_layer.py [hunk 2]
```diff
@@ -181,12 +178,12 @@ def fit_transform(self, X, y, sample_weight=None):
         self.oob_decision_function_ = oob_decision_function / self.n_estimators
         if self.is_classifier:
             y_pred = np.argmax(oob_decision_function, axis=1)
-            self.val_acc_ = accuracy_score(
```
Comment (xuyxu) [inline deepforest/_layer.py:184]: See comments above.

**Review by xuyxu — COMMENTED** [review]
Hi @Alex-Medium, the codes look good. I have an additional suggestion, please refer to the comment for details.

**Comment by xuyxu** [comment]
Thanks @Alex-Medium, we are closer to the merge ;-). In addition, could you modify the docstrings of `CascadeForestRegressor`, here is a good example: [RandomForestRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html).

**Review by xuyxu — COMMENTED** [review]
LGTM.

**Comment by xuyxu** [comment]
Looks all good. I will merge this PR after conducting some experiments.

**Comment by xuyxu** [comment]
@all-contributors please add @Alex-Medium for code,test

**Comment by xuyxu** [comment]
@all-contributors please add @Alex-Medium for code test

**Comment by xuyxu** [comment]
**Testing mse on [Sarcos](http://www.gaussianprocess.org/gpml/data/) dataset**:
* deep forest: 0.44325 (2 layers) | 0.35531 (1 layer)
* Random forest: 0.39799
* XGB-EXACT + Scikit-Learn.MultiOutputRegressor: 1.37665
* XGB-HIST + Scikit-Learn.MultiOutputRegressor: 1.34101
* LightGBM + Scikit-Learn.MultiOutputRegressor: 2.62116

The experiment result looks promising, at least much better than GBDTs. However, multiple cascade layers seem to deteriorate the performance on this dataset, maybe a furture work to work on.

``` python
import scipy.io as scio
from sklearn.metrics import mean_squared_error

from deepforest import CascadeForestRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.multioutput import MultiOutputRegressor
from xgboost.sklearn import XGBRegressor
from lightgbm import LGBMRegressor


if __name__ == "__main__":
    
    train = scio.loadmat("sarcos_inv.mat")["sarcos_inv"]
    test = scio.loadmat("sarcos_inv_test.mat")["sarcos_inv_test"]

    X_train, y_train = train[:, :21], train[:, 21:]
    X_test, y_test = test[:, :21], test[:, 21:]

    model = CascadeForestRegressor(n_jobs=-1, verbose=2, random_state=0)
    model.fit(X_train, y_train)
    y_pred = model.predict(X_test)
    print("Testing MSE: {:.5f}".format(mean_squared_error(y_test, y_pred)))
    
    model = RandomForestRegressor(n_estimators=800, n_jobs=-1, random_state=0)
    model.fit(X_train, y_train)
    y_pred = model.predict(X_test)
    print("Testing MSE: {:.5f}".format(mean_squared_error(y_test, y_pred)))

    single = XGBRegressor(n_estimators=800 // 7,
                          objective="reg:squarederror",
                          tree_method='exact',
                          n_jobs=-1)
    model = MultiOutputRegressor(single)
    model.fit(X_train, y_train)
    y_pred = model.predict(X_test)
    print("Testing MSE: {:.5f}".format(mean_squared_error(y_test, y_pred)))

    single = XGBRegressor(n_estimators=800 // 7,
                          objective="reg:squarederror",
                          tree_method='hist',
                          n_jobs=-1)
    model = MultiOutputRegressor(single)
    model.fit(X_train, y_train)
    y_pred = model.predict(X_test)
    print("Testing MSE: {:.5f}".format(mean_squared_error(y_test, y_pred)))

    single = LGBMRegressor(boosting_type='gbdt',
                            n_estimators=800 // 7,
                            n_jobs=-1)
    model = MultiOutputRegressor(single)
    model.fit(X_train, y_train)
    y_pred = model.predict(X_test)
    print("Testing MSE: {:.5f}".format(mean_squared_error(y_test, y_pred)))
```

**Pull request merged**